### PR TITLE
[vektiva] code improvements based on suggestions from review

### PR DESCRIPTION
**Pull request by octa22**
Implemented some code improvements based on @J-N-K 's review

https://github.com/openhab/openhab2-addons/pull/4626

### CODE REVIEW

File: bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/VektivaHandlerFactory.java [hunk 1]
```diff
@@ -70,20 +77,10 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         return null;
     }
 
-    @Reference
-    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
-        this.httpClient = httpClientFactory.getCommonHttpClient();
-    }
-
     protected void unsetHttpClientFactory(HttpClientFactory httpClientFactory) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/VektivaHandlerFactory.java]: You can also remove the unset-methods

File: bundles/org.openhab.binding.vektiva/src/main/resources/ESH-INF/thing/smarwi.xml [hunk 2]
```diff
@@ -15,7 +15,7 @@
 		</channels>
 
 		<config-description>
-			<parameter name="ip" type="text" required="true">
```
Comment (J-N-K) [inline bundles/org.openhab.binding.vektiva/src/main/resources/ESH-INF/thing/smarwi.xml]: I think you removed the wrong required=„true“ here. IP is required, refresh not.

Comment (octa22) [reply bundles/org.openhab.binding.vektiva/src/main/resources/ESH-INF/thing/smarwi.xml]: you are right. reverted.

File: bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/VektivaHandlerFactory.java (hunk 1 above)
Comment (octa22) [reply bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/internal/VektivaHandlerFactory.java]: ok done.

**Pull request merged**